test/solidlsp/scala/test_scala_language_server.py

```diff
@@ -0,0 +1,83 @@
+import logging
+import os
+
+import pytest
+
+from solidlsp.language_servers.scala_language_server import ScalaLanguageServer
+from solidlsp.ls_config import Language, LanguageServerConfig
+from solidlsp.ls_logger import LanguageServerLogger
+from solidlsp.settings import SolidLSPSettings
+
+RELATIVE_FILE_PATH = "src/main/scala/Main.scala"
+MAIN_FILE_PATH = "src/main/scala/Main.scala"
+TEST_POSITIONS = [
+    (18, 15),
+    (18, 12),
+    (19, 15),
+]
+EXPECTED_FILE_PATTERN = "Main.scala"
+EXPECTED_SYMBOLS = ["Main", "main", "add", "processUser"]
+EXPECTED_FILE_PATTERNS = ["Utils.scala", "Main.scala"]
+
+
+@pytest.fixture(scope="module")
+def scala_ls():
+    repo_root = os.path.abspath("test/resources/repos/scala")
+    logger = LanguageServerLogger(json_format=False, log_level=logging.INFO)
+    config = LanguageServerConfig(code_language=Language.SCALA)
+    solidlsp_settings = SolidLSPSettings()
+    ls = ScalaLanguageServer(config, logger, repo_root, solidlsp_settings)
+
+    with ls.start_server():
+        yield ls
+
+
+def test_scala_document_symbols(scala_ls):
+    """Test document symbols for Main.scala"""
+    symbols, _ = scala_ls.request_document_symbols(RELATIVE_FILE_PATH)
+    symbol_names = [s["name"] for s in symbols]
+
+    assert symbol_names[0] == "com.example"
+    assert symbol_names[1] == "Main"
+    assert symbol_names[2] == "main"
+    assert symbol_names[7] == "add"
+    assert symbol_names[8] == "processUser"
+
+    assert symbols[0]["kind"] == 4
+    assert symbols[1]["kind"] == 2
+    assert symbols[2]["kind"] == 6
+    assert symbols[7]["kind"] == 6
+    assert symbols[8]["kind"] == 6
+
+
+def test_scala_references_within_same_file(scala_ls):
+    """Test finding references within the same file."""
+    references_results = []
+    for line, char in TEST_POSITIONS:
+        refs = scala_ls.request_references(RELATIVE_FILE_PATH, line, char)
+        references_results.append(refs)
+
+    assert references_results[0] == []
+    assert references_results[1] == []
+    assert len(references_results[2]) == 1
+
+    first_ref = references_results[2][0]
+    assert first_ref["uri"].endswith("Main.scala")
+    assert first_ref["range"]["start"]["line"] == 19
+    assert first_ref["range"]["start"]["character"] == 14
+    assert first_ref["range"]["end"]["line"] == 19
+    assert first_ref["range"]["end"]["character"] == 17
+
+
+def test_scala_references_across_files_utils_multiply(scala_ls):
```

Comment:
this is not testing finding across files, the only reference found is within the same file as the symbol it's referring to. Pls fix this by requesting references to something that is actually referenced in another file and make sure the referencing symbol is found

Reply:
https://github.com/oraios/serena/pull/528/commits/a90a80ca2924763aec11d0c3fee5ff7d59cab996

Reply:
Sorry. I pushed something that had not been properly tested due to a local issue.